integrations/integration.php

```diff
@@ -1,57 +1,261 @@
+ */
+abstract class Site_Integration_Status {
+	const ENABLED  = 'enabled';
+	const DISABLED = 'disabled';
+	const BLOCKED  = 'blocked';
+}
+
 /**
  * Abstract base class for all integration implementations.
  *
  * @private
  */
 abstract class Integration {
+	/**
+	 * Slug of the integration.
+	 *
+	 * @var string
+	 */
+	private string $slug;
+
 	/**
 	 * An optional configuration array for this integration, added during activation.
 	 *
 	 * @var array
 	 */
-	protected array $config = [];
+	private array $config = [];
 
 	/**
-	 * A boolean indicating if this integration should be loaded. Defaults to false.
+	 * Configurations provided by VIP for setup.
+	 *
+	 * @var array {
+	 *   'client'        => array<string, string>,
+	 *   'site'          => array<string, mixed>,
+	 *   'network_sites' => array<string, array<string, mixed>>,
+	 * }
+	 *
+	 * @example
+	 * array(
+	 *  'client'        => array( 'status' => 'blocked' ),
+	 *  'site'          => array(
+	 *      'status' => 'enabled',
+	 *      'configs' => array(),
+	 *   ),
+	 *  'network_sites' => array (
+	 *      1 => array (
+	 *          'status' => 'disabled',
+	 *          'configs' => array(),
+	 *      ),
+	 *      2 => array (
+	 *          'status' => 'enabled',
+	 *          'configs' => array(),
+	 *      ),
+	 *      3 => array (
+	 *          'status' => 'blocked',
+	 *          'configs' => array(),
+	 *      ),
+	 *  )
+	 * );
+	 */
+	private array $vip_config = [];
+
+	/**
+	 * A boolean indicating if this integration is activated by customer.
 	 *
 	 * @var bool
 	 */
-	protected bool $is_active = false;
+	public bool $is_active_by_customer = false;
+
+	/**
+	 * Name of the filter which we will use to setup the integration configs.
+	 *
+	 * As of now there is no default so each integration will define its own filter in their class.
+	 *
+	 * @var string
+	 */
+	protected string $integration_configs_filter_name = '';
+
+	/**
+	 * Constructor.
+	 *
+	 * @param string $slug Slug of the integration.
+	 */
+	public function __construct( string $slug ) {
+		$this->slug = $slug;
+		
+		$this->set_vip_config();
+	}
 
 	/**
 	 * Activates this integration with an optional configuration value.
 	 *
-	 * @param array  $config An associative array of configuration values for the integration.
+	 * @param array $config An associative array of configuration values for the integration.
 	 *
 	 * @private
 	 */
 	public function activate( array $config = [] ): void {
-		$this->is_active = true;
-		$this->config    = $config;
+		$this->is_active_by_customer = true;
+		$this->config                = $config;
 	}
 
 	/**
 	 * Returns true if this integration has been activated.
 	 *
+	 * @return bool
+	 *
 	 * @private
 	 */
 	public function is_active(): bool {
-		return $this->is_active;
+		if ( $this->is_active_by_customer ) {
+			return true;
+		}
+
+		if ( $this->is_active_by_vip() ) {
+			return true;
+		}
+
+		return false;
```

Comment:
Would simplifying this down to

```suggestion
		return ( $this->is_active_by_customer ) || ( $this->is_active_by_vip() );
```

work?

Reply:
In my recent PR's mo always suggested to not have conditions in same line to avoid complexity and makes code easier to read which I somewhat agrees (not making this change)